crates/weaver_health/src/attribute_advice.rs

```diff
@@ -0,0 +1,308 @@
+                    message: "Namespace matches existing attribute".to_owned(),
+                    advisory: Advisory::Violation,
+                });
+            }
+
+            // Extends an existing namespace
+            if let Some(existing_namespace) = health_checker.find_namespace(&namespace) {
+                return Some(Advice {
+                    key: "namespace".to_owned(),
+                    value: Value::String(existing_namespace),
+                    message: "Extends existing namespace".to_owned(),
+                    advisory: Advisory::Information,
+                });
+            }
+        } else {
+            // Does not have a namespace
+            return Some(Advice {
+                key: "namespace".to_owned(),
+                value: Value::Bool(false),
+                message: "Does not have a namespace".to_owned(),
+                advisory: Advisory::Improvement,
+            });
+        }
+
+        None
+    }
+}
+
+/// An advisor that checks if an attribute has the correct type
+pub struct TypeAdvisor;
+impl Advisor for TypeAdvisor {
+    fn advise(
+        &self,
+        attribute: &SampleAttribute,
+        _health_checker: &AttributeHealthChecker,
+        semconv_attribute: Option<&Attribute>,
+    ) -> Option<Advice> {
+        // Only provide advice if the attribute is a match and the type is present
+        match (semconv_attribute, attribute.r#type.as_ref()) {
+            (Some(semconv_attribute), Some(attribute_type)) => {
+                let semconv_attribute_type = match &semconv_attribute.r#type {
+                    AttributeType::PrimitiveOrArray(primitive_or_array_type_spec) => {
+                        primitive_or_array_type_spec
+                    }
+                    AttributeType::Template(template_type_spec) => &match template_type_spec {
+                        TemplateTypeSpec::Boolean => PrimitiveOrArrayTypeSpec::Boolean,
+                        TemplateTypeSpec::Int => PrimitiveOrArrayTypeSpec::Int,
+                        TemplateTypeSpec::Double => PrimitiveOrArrayTypeSpec::Double,
+                        TemplateTypeSpec::String => PrimitiveOrArrayTypeSpec::String,
+                        TemplateTypeSpec::Strings => PrimitiveOrArrayTypeSpec::Strings,
+                        TemplateTypeSpec::Ints => PrimitiveOrArrayTypeSpec::Ints,
+                        TemplateTypeSpec::Doubles => PrimitiveOrArrayTypeSpec::Doubles,
+                        TemplateTypeSpec::Booleans => PrimitiveOrArrayTypeSpec::Booleans,
+                    },
+                    AttributeType::Enum { .. } => {
+                        // Special case: Enum variants can be either string or int
+                        if attribute_type != &PrimitiveOrArrayTypeSpec::String
+                            && attribute_type != &PrimitiveOrArrayTypeSpec::Int
+                        {
+                            return Some(Advice {
+                                key: "type".to_owned(),
```

Comment:
Perhaps `type_mismatch` would be more precise.

Reply:
Yes. I will make all the keys better. https://github.com/open-telemetry/weaver/pull/630#discussion_r1997671662

Reply:
See https://github.com/open-telemetry/weaver/pull/630/commits/7a86e4c5ccc32d00a5bef9265eb629e285f7a4c0